lib/oxide-vpc/src/engine/gateway/icmpv6.rs

```diff
@@ -89,17 +86,32 @@ pub fn setup(
             true,
             // Respond to solicitations from `::`
             true,
-        )));
-    rule_actions.push(neighbor_advert);
-
-    let n_rule_actions = rule_actions.len();
+        ))),
+    ];
 
+    // UNWRAP SAFETY: There are far fewer than 65535 rules inserted here.
+    let next_out_prio = u16::try_from(hairpins.len() + 1).unwrap();
     // Add rules for the above actions.
-    for i in 0..n_rule_actions {
+    hairpins.into_iter().enumerate().for_each(|(i, action)| {
         let priority = u16::try_from(i + 1).unwrap();
-        let rule = Rule::new(priority, rule_actions.remove(0));
+        let rule = Rule::new(priority, action);
         layer.add_rule(Direction::Out, rule.finalize());
-    }
+    });
+
+    // Filter any uncaught in/out-bound NDP traffic.
```

Comment:
I suppose it's not a bad idea to have the rules, but isn't the gateway layer deny by default?

Reply:
The layer is deny by default, but due to the [last in/outbound rules applied here](https://github.com/oxidecomputer/opte/blob/master/lib/oxide-vpc/src/engine/gateway/mod.rs#L211-L234) this only denies IP-spoofed or misaddressed packets. The NDP packets in question have the correct unicast sender/recipient, so would not be denied.